pkgs/racket-test-core/tests/racket/procs.rktl

```diff
@@ -448,6 +448,10 @@
 
 (test -4 procedure-arity-mask apply)
 
+;; ------------------------------------------------------------
+;; Check if `lambda` and `λ` access the same binding.
+(test #t free-identifier=? #'lambda #'λ)
+
```

Comment:
The `syntax.rktl` file is probably a better place for this check, since that one is more about syntactic forms, binding, and syntax checking. But I don't object if you really prefer the test here.